webSites/webSites-war/src/main/java/org/silverpeas/components/websites/servlets/WebSitesRequestRouter.java

```diff
@@ -90,1056 +87,1137 @@ public WebSiteSessionController createComponentSessionController(
+      SiteDetail ancien = scc.getWebSite(id);
+      int type = ancien.getType();
 
           /* update description en BD */
-          SiteDetail descriptionSite2 =
-              new SiteDetail(id, scc.getComponentId(), nomSite, description, nomPage, type, null,
-                  null, etat, popup);
+      SiteDetail descriptionSite2 =
+          new SiteDetail(id, scc.getComponentId(), nomSite, description, nomPage, type, null,
+              null, etat, popup);
 
-          scc.updateWebSite(descriptionSite2);
+      scc.updateWebSite(descriptionSite2);
 
-          if (listIcons.size() > 0) {
-            scc.associateIcons(id, listIcons);
-          }
+      if (listIcons.size() > 0) {
+        scc.associateIcons(id, listIcons);
+      }
 
           /* publications : classer le site dans les themes cochés */
-          ArrayList<String> arrayToClassify = new ArrayList<>();
-          boolean publish = false;
-          ArrayList<String> arrayTopic = new ArrayList<>();
-          begin = 0;
-          end = 0;
-          end = listeTopics.indexOf(',', begin);
-          String idTopic;
-          while (end != -1) {
-            idTopic = listeTopics.substring(begin, end);
+      ArrayList<String> arrayToClassify = new ArrayList<>();
+      boolean publish = false;
+      ArrayList<String> arrayTopic = new ArrayList<>();
+      begin = 0;
+      end = 0;
+      end = listeTopics.indexOf(',', begin);
+      String idTopic;
+      while (end != -1) {
+        idTopic = listeTopics.substring(begin, end);
+
+        begin = end + 1;
+        end = listeTopics.indexOf(',', begin);
+
+        arrayTopic.add(idTopic);
+        publish = true;
+      }
 
-            begin = end + 1;
-            end = listeTopics.indexOf(',', begin);
+      scc.updateClassification(id, arrayTopic);
 
-            arrayTopic.add(idTopic);
-            publish = true;
-          }
+      arrayToClassify.add(id);
+      if (publish) {
+        scc.publish(arrayToClassify); // set etat du site a 1
+      } else {
+        scc.dePublish(arrayToClassify);
+      }
+    }
 
-          scc.updateClassification(id, arrayTopic);
+    Collection<SiteDetail> listeSites = scc.getAllWebSite();
+    request.setAttribute("ListSites", listeSites);
+    request.setAttribute("BookmarkMode", Boolean.valueOf(scc.isBookmarkMode()));
 
-          arrayToClassify.add(id);
-          if (publish) {
-            scc.publish(arrayToClassify); // set etat du site a 1
-          } else {
-            scc.dePublish(arrayToClassify);
-          }
-        }
+    return "/webSites/jsp/manage.jsp";
+  }
 
-        Collection<SiteDetail> listeSites = scc.getAllWebSite();
-        request.setAttribute("ListSites", listeSites);
-        request.setAttribute("BookmarkMode", Boolean.valueOf(scc.isBookmarkMode()));
+  private String classification(final WebSiteSessionController scc, final HttpRequest request,
+      final String userRole) throws WebSitesException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused method parameter "userRole". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1172)